lib/src/value/calculation.dart

```diff
@@ -158,6 +382,248 @@ class SassCalculation extends Value {
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value log(Object number, Object? base) {
+    number = _simplify(number);
+    base = base.andThen(_simplify);
+    var args = [number, if (base != null) base];
+    if (number is! SassNumber ||
+        (base != null && base is! SassString && base is! SassNumber) ||
+        number.hasUnit('%') ||
+        (base is SassNumber && base.hasUnit('%'))) {
+      return SassCalculation._("log", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return number_lib.log(number, base is SassNumber ? base : null);
+  }
+
+  /// Creates a `atan2()` calculation for [y] and [x].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value atan2(Object y, Object? x) {
+    _verifyLength([y, if (x != null) x], 2);
+    y = _simplify(y);
+    x = x.andThen(_simplify);
+    var args = [y, if (x != null) x];
+    if (y is! SassNumber ||
+        x is! SassNumber ||
+        y.hasUnit('%') ||
+        x.hasUnit('%')) {
+      return SassCalculation._("atan2", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return number_lib.atan2(y, x);
+  }
+
+  /// Creates a `rem()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value rem(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        dividend.hasUnit('%') ||
+        modulus.hasUnit('%')) {
+      return SassCalculation._("rem", args);
+    }
+    _verifyCompatibleNumbers(args);
+    if (modulus.value.isInfinite) return dividend;
+    var modulo = dividend.modulo(modulus);
+    return (modulo is SassNumber && modulo.value == 0)
+        ? SassNumber(-0.0)
+        : modulo.minus(dividend);
```

Comment:
`rem()`'s implementation should match the spec here, so you should calculate `var result = dividend.modulo(modulus)` first, and then check for the special cases only if `dividend` and `modulus` have different signs.

Reply:
Oh yes, checking the different signs was missing here. Thanks!